Source/EasyNetQ/Consumer/InternalConsumer.cs

```diff
@@ -82,7 +84,7 @@ public interface IInternalConsumer : IDisposable
     /// <inheritdoc />
     public class InternalConsumer : IInternalConsumer
     {
-        private readonly Dictionary<IQueue, AsyncBasicConsumer> consumers = new Dictionary<IQueue, AsyncBasicConsumer>();
+        private readonly ConcurrentDictionary<string, AsyncBasicConsumer> consumers = new ConcurrentDictionary<string, AsyncBasicConsumer>();
```

Comment:
That's ok, but we are mixing different context in a single PR, I was looking for drop interfaces, and I'm checking also concurrency and other stuff like `ExchangeTo*Binding`

Reply:
I've reverted this stuff